test/WebJobs.Script.Tests.Integration/Storage/AzureStorageProviderTests.cs

```diff
@@ -0,0 +1,234 @@
+                await VerifyContainerClientAvailable(container);
+
+                Assert.True(azureStorageProvider.TryGetBlobServiceClientFromConnection(out BlobServiceClient blobServiceClient, ConnectionStringNames.Storage));
+                await VerifyBlobServiceClientAvailable(blobServiceClient);
+            }
+        }
+
+        [Fact]
+        public void TestAzureStorageProvider_NoConnectionThrowsException()
+        {
+            var testData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "AzureWebJobsStorage", "" }
+            };
+
+            using (new TestScopedEnvironmentVariable(testData))
+            {
+                var webHostConfiguration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddTestSettings()
+                    .Build();
+
+                var jobHostConfiguration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddTestSettings()
+                    .Build();
+
+                var azureStorageProvider = GetAzureStorageProvider(webHostConfiguration, jobHostConfiguration);
+                Assert.Throws<InvalidOperationException>(() => azureStorageProvider.GetBlobContainerClient());
+
+                Assert.False(azureStorageProvider.TryGetBlobServiceClientFromConnection(out BlobServiceClient blobServiceClient, ConnectionStringNames.Storage));
+            }
+        }
+
+        [Theory]
+        [InlineData("ConnectionStrings:AzureWebJobsStorage1")]
+        [InlineData("AzureWebJobsStorage1")]
+        [InlineData("Storage1")]
+        public void TestAzureStorageProvider_JobHostConfigurationWinsConflict(string connectionName)
+        {
+            var bytes = Encoding.UTF8.GetBytes("someKey");
+            var encodedString = Convert.ToBase64String(bytes);
+
+            var testData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                { connectionName, $"DefaultEndpointsProtocol=https;AccountName=webHostAccount;AccountKey={encodedString};EndpointSuffix=core.windows.net" },
+            };
+
+            using (new TestScopedEnvironmentVariable(testData))
+            {
+                var webHostConfiguration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddTestSettings()
+                    .Build();
+
+                var inMemory = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+                {
+                    { connectionName, $"DefaultEndpointsProtocol=https;AccountName=jobHostAccount;AccountKey={encodedString};EndpointSuffix=core.windows.net" },
+                };
+                var jobHostConfiguration = new ConfigurationBuilder()
+                    .AddInMemoryCollection(inMemory)
+                    .AddTestSettings()
+                    .Build();
```

Comment:
Do you want `AddTestSettings()` here? That adds a test settings file that we support for tests. If you don't have one, this test will pass, but if you do it may fail. May just want to remove `AddTestSettings()` completely here (and elsewhere that you use this pattern).

Reply:
I removed AddTestSettings in this file since we should only be using InMemoryCollection anyways.